diff --git a/lib/src/modules/dex/parser.rs b/lib/src/modules/dex/parser.rs
index 7a652038..7642c289 100644
--- a/lib/src/modules/dex/parser.rs
+++ b/lib/src/modules/dex/parser.rs
@@ -440,7 +440,7 @@ impl Dex {
                     class_idx,
                     access_flags,
                     superclass_idx,
-                    _,
+                    interfaces_off,
                     source_file_idx,
                     _,
                     _,
@@ -461,6 +461,7 @@ impl Dex {
                     let mut item = protos::dex::ClassItem::new();
                     item.class = Some(class);
                     item.set_access_flags(access_flags);
+                    item.set_interfaces_off(interfaces_off);
                     if let Some(superclass) = superclass {
                         item.superclass = Some(superclass);
                     }
@@ -605,6 +606,21 @@ impl From<DexHeader> for protos::dex::DexHeader {
         result.set_endian_tag(header.endian_tag);
         result.set_link_size(header.link_size);
         result.set_link_off(header.link_off);
+        result.set_map_off(header.map_off);
+        result.set_string_ids_size(header.string_ids_size);
+        result.set_string_ids_off(header.string_ids_off);
+        result.set_type_ids_size(header.type_ids_size);
+        result.set_type_ids_off(header.type_ids_off);
+        result.set_proto_ids_size(header.proto_ids_size);
+        result.set_proto_ids_off(header.proto_ids_off);
+        result.set_field_ids_size(header.field_ids_size);
+        result.set_field_ids_off(header.field_ids_off);
+        result.set_method_ids_size(header.method_ids_size);
+        result.set_method_ids_off(header.method_ids_off);
+        result.set_class_defs_size(header.class_defs_size);
+        result.set_class_defs_off(header.class_defs_off);
+        result.set_data_size(header.data_size);
+        result.set_data_off(header.data_off);
         result.container_size = header.container_size;
         result.header_offset = header.header_offset;
         result
