diff --git a/lib/src/modules/protos/dex.proto b/lib/src/modules/protos/dex.proto
index acb238c4..a85f5c29 100644
--- a/lib/src/modules/protos/dex.proto
+++ b/lib/src/modules/protos/dex.proto
@@ -50,6 +50,19 @@ message DexHeader {
     optional uint32 link_size = 8;
     // Offset pointing to the link section data.
     optional uint32 link_off = 9 [(yara.field_options).fmt = "x"];
+    optional uint32 map_off = 10 [(yara.field_options).fmt = "x"];
+    optional uint32 string_ids_size = 11;
+    optional uint32 string_ids_off = 12 [(yara.field_options).fmt = "x"];
+    optional uint32 type_ids_size = 13;
+    optional uint32 type_ids_off = 14 [(yara.field_options).fmt = "x"];
+    optional uint32 proto_ids_size = 15;
+    optional uint32 proto_ids_off = 16 [(yara.field_options).fmt = "x"];
+    optional uint32 field_ids_size = 17;
+    optional uint32 field_ids_off = 18 [(yara.field_options).fmt = "x"];
+    optional uint32 method_ids_size = 19;
+    optional uint32 method_ids_off = 20 [(yara.field_options).fmt = "x"];
+    optional uint32 class_defs_size = 21;
+    optional uint32 class_defs_off = 22 [(yara.field_options).fmt = "x"];
     // Size in bytes of the main data section.
     optional uint32 data_size = 23;
     // File offset pointing to the main data block.
@@ -96,8 +109,10 @@ message ClassItem {
     optional uint32 access_flags = 2 [(yara.field_options).fmt = "flags:AccessFlag"];
     // Superclass descriptor inherited by this object.
     optional string superclass = 3;
+    // File offset pointing to the interfaces list.
+    optional uint32 interfaces_off = 4 [(yara.field_options).fmt = "x"];
     // Source code file name metadata string.
-    optional string source_file = 4;
+    optional string source_file = 5;
 }
 
 message MapList {
