diff --git a/lib/src/modules/protos/dex.proto b/lib/src/modules/protos/dex.proto
index 8e6f6d1f..6abaa790 100644
--- a/lib/src/modules/protos/dex.proto
+++ b/lib/src/modules/protos/dex.proto
@@ -34,6 +34,19 @@ message DexHeader {
     optional uint32 endian_tag = 7 [(yara.field_options).fmt = "x"];
     optional uint32 link_size = 8;
     optional uint32 link_off = 9 [(yara.field_options).fmt = "x"];
+    optional uint32 map_off = 10 [(yara.field_options).fmt = "x"];
+    optional uint32 string_ids_size = 11;
+    optional uint32 string_ids_off = 12 [(yara.field_options).fmt = "x"];
+    optional uint32 type_ids_size = 13;
+    optional uint32 type_ids_off = 14 [(yara.field_options).fmt = "x"];
+    optional uint32 proto_ids_size = 15;
+    optional uint32 proto_ids_off = 16 [(yara.field_options).fmt = "x"];
+    optional uint32 field_ids_size = 17;
+    optional uint32 field_ids_off = 18 [(yara.field_options).fmt = "x"];
+    optional uint32 method_ids_size = 19;
+    optional uint32 method_ids_off = 20 [(yara.field_options).fmt = "x"];
+    optional uint32 class_defs_size = 21;
+    optional uint32 class_defs_off = 22 [(yara.field_options).fmt = "x"];
     optional uint32 data_size = 23;
     optional uint32 data_off = 24 [(yara.field_options).fmt = "x"];
     optional uint32 container_size = 25;
@@ -63,7 +76,8 @@ message ClassItem {
     optional string class = 1;
     optional uint32 access_flags = 2 [(yara.field_options).fmt = "flags:AccessFlag"];
     optional string superclass = 3;
-    optional string source_file = 4;
+    optional uint32 interfaces_off = 4;
+    optional string source_file = 5;
 }
 
 message MapList {
